File: DNN.py
import tensorflow.keras.backend as K
from tensorflow.keras.layers import Input,Dense,BatchNormalization,Activation,Conv1D,\
    Flatten,Reshape,Dropout,concatenate,MaxPooling1D,add
from tensorflow.keras import Model
from tensorflow.keras import regularizers

def DNN(input_dim,layer_list,activation="relu",dropout=0.2):
    input=Input((input_dim,))
    for i in range(len(layer_list)):
        if i==0:
            dense=Dense(layer_list[i],activation=activation,
                        kernel_regularizer=regularizers.l2(0.001))(input)
            dense=BatchNormalization()(dense)
            dense=Activation(activation)(dense)
            if dropout>0:
                dense=Dropout(dropout)(dense)
        else:
            dense=Dense(layer_list[i],activation=activation,
                        kernel_regularizer=regularizers.l2(0.001))(dense)
            dense=BatchNormalization()(dense)
            dense=Activation(activation)(dense)
            if dropout>0:
                dense=Dropout(dropout)(dense)
    output=Dense(1)(dense)
    DNN_model=Model(input,output)
    return DNN_model

File: DNN.R
library(keras)
tensorflow::tf$random$set_seed(123)
DNN<-function(input_dim,layer_list,dropout=0.2,activation="relu"){
  input<-layer_input(shape=c(input_dim))
  for (i in 1:length(layer_list)){
    if(i==1){
      dense<-input%>%layer_dense(units=layer_list[i],activation =activation, kernel_regularizer = regularizer_l2(l = 0.001))%>%
        layer_batch_normalization()%>%
        layer_activation( activation)
      if(dropout>0){
        dense<-dense%>%layer_dropout(dropout)
      }
    }else{
      dense<-dense%>%layer_dense(units=layer_list[i],activation =activation, kernel_regularizer = regularizer_l2(l = 0.001))%>%
        layer_batch_normalization()%>%
        layer_activation( activation)
      if(dropout>0){
        dense<-dense%>%layer_dropout(dropout)
      }
    }
  }
  output<-dense%>%layer_dense(units=1,activation = NULL)
  DNN_model<-keras_model(input=input,output=output)
  DNN_model%>%compile(    loss = 'mse',
                          optimizer = optimizer_adam())
  DNN_model
}